### PR TITLE
WhoLikedYou and Navigation

### PR DESCRIPTION
**Pull request by nasnik**
1. Introduced the YouLikedList component, which shares similarities with the NearMe component. To enhance code reusability, I restructured the project folders: the "helpers" folder was relocated to the /src directory, and a common "Lightning" component was created to provide a lightning badge.

2. Implemented navigation with color changes for both the web app's Header and the FooterAppBar on web mobile screens, transitioning from grey to red.

The YouLikedList component can be seen at http://localhost:3000/whoLikedYou

### CODE REVIEW

File: src/components/footerAppBar/FooterAppBar.tsx [hunk 1]
```diff
@@ -2,33 +2,84 @@ import React from 'react'
...
+              )}
+            </Link>
+          }
         />
         <BottomNavigationAction
-          label="Favorites"
-          icon={<img src="img/nearBy/like_you.svg" />}
+          value="wholikedyou"
+          icon={
+            <Link to="/whoLikedYou" style={{ textDecoration: 'none' }}>
```
Comment (LyubovSugak) [inline src/components/footerAppBar/FooterAppBar.tsx:35]: Same here, it's hard to read code

File: src/components/footerAppBar/FooterAppBar.tsx [hunk 2]
```diff
@@ -2,33 +2,84 @@ import React from 'react'
...
+            </Link>
+          }
         />
         <BottomNavigationAction
-          label="Archive"
-          icon={<img src="img/nearBy/ii.svg" />}
+          value="friends"
+          icon={
+            <Link to="/messages" style={{ textDecoration: 'none' }}>
+              {activePage === 'friends' ? (
```
Comment (LyubovSugak) [inline src/components/footerAppBar/FooterAppBar.tsx:54]: Same here

File: src/components/footerAppBar/FooterAppBar.tsx
Comment (LyubovSugak) [inline src/components/footerAppBar/FooterAppBar.tsx:1]: I would suggest to think about refactoring all those conditions. They are similar. Maybe to store like key/value

File: src/components/header/Header.tsx
Comment (LyubovSugak) [inline src/components/header/Header.tsx:1]: The same as in Footer component regards of images. Should be a helper with option of width/height, etc

File: src/components/nearMe/UserListRenderer.tsx [hunk 3]
```diff
@@ -32,7 +34,8 @@ const UserListRenderer: React.FC<UserListRendererProps> = ({
                 className={classes.userImages}
                 alt="user profile"
               />
-              <Box sx={{ fontWeight: 'bold' }}>
+              {user.likedUsers.includes(currentUserID) && <Lightning />}
```
Comment (LyubovSugak) [inline src/components/nearMe/UserListRenderer.tsx:37]: I mentioned already, please don't add logic to html. Filter, includes whatever should be above. For better understanding - please read about clean code.

**Review by LyubovSugak — CHANGES_REQUESTED** [review]

File: src/components/footerAppBar/FooterAppBar.tsx [hunk 4]
```diff
@@ -2,33 +2,84 @@ import React from 'react'
...
+        value={activePage}
+        onChange={(event, newValue) => setNewActivePage(newValue)}
+      >
         <BottomNavigationAction
-          label="Recents"
-          icon={<img src="/img/nearBy/near_me.svg" />}
+          value="nearme"
+          icon={
+            <Link to="/nearMe" style={{ textDecoration: 'none' }}>
+              {activePage === 'nearme' ? (
```
[the remark this answers is not in the document]
Comment (nasnik) [reply src/components/footerAppBar/FooterAppBar.tsx]: Refactored components Header.tsx and FooterAppBar.tsx to get rid of conditionals. Created the following helpers: navigationConfig.ts, navigationRenderer.tsx, renderIcon.ts.

File: src/components/footerAppBar/FooterAppBar.tsx (hunk 1 above)
Comment (nasnik) [reply src/components/footerAppBar/FooterAppBar.tsx]: Refactored components Header.tsx and FooterAppBar.tsx to get rid of conditionals. Created the following helpers: navigationConfig.ts, navigationRenderer.tsx, renderIcon.ts.

File: src/components/footerAppBar/FooterAppBar.tsx
Comment (nasnik) [reply src/components/footerAppBar/FooterAppBar.tsx:1]: Refactored components Header.tsx and FooterAppBar.tsx to get rid of conditionals. Created the following helpers: navigationConfig.ts, navigationRenderer.tsx, renderIcon.ts.

File: src/components/nearMe/UserListRenderer.tsx (hunk 3 above)
Comment (nasnik) [reply src/components/nearMe/UserListRenderer.tsx:37]: "This string with currentUserID is included in this special renderer file temporarily, pending the completion of the backend. It serves as a placeholder until the backend functionality is ready."

**Comment by LyubovSugak** [comment]
@nasnik Please revert package-lock. Use npm ci instead install. If there are no chnages in package.json, package-lock should be not updated

File: src/components/lightning/Lightning.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+
+const Lightning = () => {
+  return (
+    <img
+      src="/img/whoLikesMe/who_liked_me_red.svg"
+      alt="Lightning"
+      style={{
```
Comment (LyubovSugak) [inline src/components/lightning/Lightning.tsx:8]: plz move style to according hook

File: src/helpers/navigationRenderer.tsx [hunk 6]
```diff
@@ -0,0 +1,47 @@
+import React from 'react'
+import { Link } from 'react-router-dom'
+import BottomNavigationAction from '@mui/material/BottomNavigationAction'
+import RenderIcon from './renderIcon'
+
+interface NavigationItem {
+  value: string
+  iconProps: any // Adjust the type according to your needs
```
Comment (LyubovSugak) [inline src/helpers/navigationRenderer.tsx:8]: type iconProps

File: src/components/nearMe/NearMeProfiles.tsx [hunk 7]
```diff
@@ -2,17 +2,17 @@ import React, { useEffect, useState } from 'react'
...
-import { UserNearMeObjectType } from './userTypes'
+import { UserObjectType } from '../../helpers/userTypes'
 type IsMobileProps = {
   isMobile: boolean
 }
 
 const NearMeProfiles = ({ isMobile }: IsMobileProps) => {
   const { classes } = useStyles()
-  const [list, setList] = useState<Array<UserNearMeObjectType>>([])
+  const [list, setList] = useState<Array<UserObjectType>>([])
```
[the remark this answers is not in the document]
Comment (nasnik) [reply src/components/nearMe/NearMeProfiles.tsx]: changed [list, setList] => [profileList, setProfileList]

File: src/components/yourLikesList/YouLikesProfiles.tsx [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+import { getColumns } from '../../helpers/helper'
+import UserListRenderer from './YourLikesUserListRenderer'
+import { UserObjectType } from '../../helpers/userTypes'
+type IsMobileProps = {
+  isMobile: boolean
+}
+
+const YouLikesProfiles = ({ isMobile }: IsMobileProps) => {
+  const { classes } = useStyles()
+  const [list, setList] = useState<Array<UserObjectType>>([])
```
[the remark this answers is not in the document]
Comment (nasnik) [reply src/components/yourLikesList/YouLikesProfiles.tsx]: changed [list, setList] => [profileList, setProfileList]

File: src/components/yourLikesList/YourLikesUserListRenderer.tsx [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+            sx={{
+              display: 'flex',
+              flexDirection: 'column',
+              justifyContent: 'center',
+              position: 'relative',
+            }}
+          >
+            <img
+              src={user.picture}
+              className={classes.userImages}
```
[the remark this answers is not in the document]
Comment (nasnik) [reply src/components/yourLikesList/YourLikesUserListRenderer.tsx:36]: Это стилизация в Material UI const { classes } = useStyles() - эту модель я взяла из компонента Header, который был единственный готов в момент моего прихода на проект, и я его взяла за пример.

File: src/helpers/navigationRenderer.tsx (hunk 6 above)
Comment (nasnik) [reply src/helpers/navigationRenderer.tsx]: In navigationRenderer.tsx changed iconProps: any=> {imagePath: string, alt: string, size: {width: number; height: number}

File: src/components/lightning/Lightning.tsx (hunk 5 above)
Comment (nasnik) [reply src/components/lightning/Lightning.tsx]: moved style to according hook in Lightning component

File: src/helpers/navigationRenderer.tsx [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+      key={value}
+      value={value}
+      icon={renderLinkWithIcon(linkTo, iconProps, size, activePage === value)}
+      onClick={() => setNewActivePage(value)}
+    />
+  ))
+}
+const renderLinkWithIcon = (
+  linkTo: string,
+  iconProps: any,
```
[the remark this answers is not in the document]
Comment (nasnik) [reply src/helpers/navigationRenderer.tsx]: In navigationRenderer.tsx changed iconProps: any=> {imagePath: string, alt: string, size: {width: number; height: number}

File: src/components/nearMe/UserListRenderer.tsx [hunk 11]
```diff
@@ -32,7 +34,8 @@ const UserListRenderer: React.FC<UserListRendererProps> = ({
...
+              <Box sx={{ fontWeight: 'bold', color: '#F46B5D' }}>
                 {user.fName} {user.lName}
```
[the remark this answers is not in the document]
Comment (nasnik) [reply src/components/nearMe/UserListRenderer.tsx]: In users.json, userTypes.ts, UserListRenderer.tsx, and YourLikesUserListRenderer.tsx changed fName=> firstName, lName => lastName.

File: src/helpers/navigationRenderer.tsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (nasnik) [reply src/helpers/navigationRenderer.tsx]: 1. Changed renderIcon.tsx and NavigationTypes.ts to get rid of errors…
… that occurred after replacing 'any' type in iconProps.

2. Created a file NavigationTypes.ts with types for Header.tsx and FooterAppBar.tsx components and their helper functions renderIcon.tsx, navigationConfigHelper.ts, and navigationRenderer.tsx.
3. Created a folder 'common' to contain common files including common types files. Transferred files with common types: NavigationTypes.ts and userTypes.ts to src/common/.

**Review by LyubovSugak — APPROVED** [review]

**Pull request merged**